File: backend/app/services/calculator.py
from app.models.schemas import CalculatorInput, CalculationResult, RepeatPeriod

def calculate(data: CalculatorInput) -> CalculationResult:
    monthly_cycles = _to_monthly_cycles(data.repeat_count, data.repeat_period)

    minutes_per_month = monthly_cycles * data.people_involved * data.minutes_per_cycle
    hours_lost_per_month = round(minutes_per_month / 60, 2)

    cost_per_month = round(hours_lost_per_month * data.hourly_rate_eur, 2)
    cost_per_year = round(cost_per_month * 12, 2)

    return CalculationResult(
        hours_lost_per_month=hours_lost_per_month,
        cost_per_month_eur=cost_per_month,
        cost_per_year_eur=cost_per_year,
    )


def _to_monthly_cycles(count: int, period: RepeatPeriod) -> float:
    if period == RepeatPeriod.daily:
        return count * 22  # working days per month
    if period == RepeatPeriod.weekly:
        return count * 4.33
    return count  # monthly

File: frontend/src/types/calculator.ts
export type BusinessSector =
  | 'trade'
  | 'services'
  | 'manufacturing'
  | 'logistics'
  | 'finance'
  | 'other';

export type EmployeeCount = '1-5' | '6-20' | '21-50' | '50+';

export type RepeatPeriod = 'daily' | 'weekly' | 'monthly';

export interface CalculatorInput {
  sector: BusinessSector;
  employee_count: EmployeeCount;
  process_description: string;
  repeat_count: number;
  repeat_period: RepeatPeriod;
  people_involved: number;
  minutes_per_cycle: number;
  hourly_rate_eur: number;
}

export interface CalculationResult {
  hours_lost_per_month: number;
  cost_per_month_eur: number;
  cost_per_year_eur: number;
}

export interface AIAnalysis {
  process_category: string;
  automation_complexity: 'Лесна' | 'Средна' | 'Сложна';
  analysis_text: string;
}

export interface CalculatorResponse {
  calculation: CalculationResult;
  ai_analysis: AIAnalysis;
}

export interface AuditRequest {
  name: string;
  email: string;
  phone?: string;
  process_description: string;
  cost_per_year_eur: number;
  automation_complexity: string;
}

export interface ContactRequest {
  name: string;
  email: string;
  phone?: string;
  message: string;
}

export type Step = 'step1' | 'step2' | 'step3' | 'results' | 'audit';